Userland/Libraries/LibWeb/Layout/LayoutState.cpp

```diff
@@ -354,9 +354,9 @@ AvailableSpace LayoutState::UsedValues::available_inner_space_or_constraints_fro
     auto inner_width = available_width_inside();
     auto inner_height = available_height_inside();
 
-    if (inner_width.is_indefinite() && outer_space.width.is_intrinsic_sizing_constraint())
+    if ((inner_width.is_indefinite() || !isfinite(inner_width.to_px().value())) && outer_space.width.is_intrinsic_sizing_constraint())
         inner_width = outer_space.width;
-    if (inner_height.is_indefinite() && outer_space.height.is_intrinsic_sizing_constraint())
```

Comment:
this check means that somewhere we have a call to construct `AvailableSize` using `make_definite(INFINITY)` which is wrong and `make_indefinite()`, `make_min_content()` or `make_max_content()` should have been used instead.

I tried to add verification that value passed in `AvailableSize::make_definite` is actually definite:
```diff
diff --git a/Userland/Libraries/LibWeb/Layout/AvailableSpace.cpp b/Userland/Libraries/LibWeb/Layout/AvailableSpace.cpp
index 3cd4004f84..1622d5db04 100644
--- a/Userland/Libraries/LibWeb/Layout/AvailableSpace.cpp
+++ b/Userland/Libraries/LibWeb/Layout/AvailableSpace.cpp
@@ -11,6 +11,7 @@ namespace Web::Layout {

 AvailableSize AvailableSize::make_definite(CSSPixels value)
 {
+    VERIFY(isfinite(value.value()));
     return AvailableSize { Type::Definite, value };
 }
 ```
 
 and it showed up that `Tests/LibWeb/Layout/input/table/in-auto-height-flex-item.html` currently constructs invalid `AvailableSize`. so now you have a reduced example to fix root cause instead of introducing a new hack! :)

Reply:
I made `compute_table_box_width_inside_table_wrapper` a bit more careful with constraints vs definite sizes, which should address your concern. There are a dozen more places where we lean on the actual value of `min-content` and `max-content` which we might want to address in the future if we want to tighten the checks more.

Reply:
thank you!